telegram/request/_httpxrequest.py

```diff
@@ -91,6 +97,15 @@ class HTTPXRequest(BaseRequest):
 
             .. versionchanged:: 20.5
                 Accept ``"2"`` as a valid value.
+        socket_options (Collection[Tuple], optional): A collection of socket options to set on
+            the underlying socket. For each entry, the first two elements must be integers
+            representing the socket level and the socket option, respectively. The third element
+            must be either an integer or a bytes-like object representing the value to set.
+            Alternatively, the entry can be a 4-tuple, where the third element is :obj:`None` and
+            the fourth element is an integer representing the timeout to set.
+            Defaults to :obj:`None`.
```

Comment:
Where was this text taken from? It sounds too confusing.

I would just go "This object is passed as is to XXX and the implementation details are OS specific. This is an advanced section, make sure you know what you're doing"

or you can say something like "see [httpx.SOCKET_OPTIONS] for more details"

Reply:
TBH I had quite a bit of help from GitHub Copilot on this :D But redirecting to httpx docs might be a saner option 👍